### PR TITLE
[CI] Add coding style check as prerequisite

### PR DESCRIPTION
**Pull request by KFilipek**
This PR introduces a coding style check.

### CODE REVIEW

**Review by vinser52 — APPROVED** [review]

File: .github/workflows/pr_push.yml [hunk 1]
```diff
@@ -48,6 +48,31 @@ jobs:
         - name: Build
           run: >
             cmake --build ${{github.workspace}}/build --config Release -j ${{matrix.nproc}}
+
+    CodeStyle:
```
Comment (PatKamin) [inline .github/workflows/pr_push.yml:52]: Add this as a requirement to the Build job.

File: .github/workflows/pr_push.yml [hunk 2]
```diff
@@ -48,6 +48,31 @@ jobs:
...
+      runs-on: ubuntu-latest
+
+      steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install apt packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y cmake clang-format-15 libjemalloc-dev libnuma-dev libtbb-dev
```
Comment (PatKamin) [inline .github/workflows/pr_push.yml:63]: Libs are redundant as you don't compile the sources, just disable building parts of umf which require these libs.

File: .github/workflows/pr_push.yml [hunk 3]
```diff
@@ -48,6 +48,31 @@ jobs:
...
+      - name: Configure CMake
+        run: >
+          cmake
+          -B ${{github.workspace}}/build
+          -DUMF_FORMAT_CODE_STYLE=ON
+
+      - name: Build
+        working-directory: ${{github.workspace}}/build
+        run: >
+          make clang-format-check
```
Comment (PatKamin) [inline .github/workflows/pr_push.yml:74]: nit: We could use cmake everywhere for code portability:
```
cmake --build build --target clang-format-check
```
You can drop the change of the working directory then.

File: .github/workflows/pr_push.yml (hunk 2 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml]: Done

File: .github/workflows/pr_push.yml (hunk 3 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml]: Done

File: .github/workflows/pr_push.yml [hunk 4]
```diff
@@ -48,6 +48,31 @@ jobs:
...
+      - name: Configure CMake
+        run: >
+          cmake
+          -B ${{github.workspace}}/build
+          -DUMF_FORMAT_CODE_STYLE=ON
+
+      - name: Build
+        run: >
+          cmake --build build --target clang-format-check  
+
```
Comment (lukaszstolarczuk) [inline .github/workflows/pr_push.yml:74]: redundant empty line

**Comment by lukaszstolarczuk** [comment]
Please add to the commit msg info about #133

`Fixes: #133`

File: .github/workflows/pr_push.yml (hunk 4 above)
Comment (lukaszstolarczuk) [inline .github/workflows/pr_push.yml:74]: I can see that clang-format-check is run on Mac (in basic.yml) - should we get rid of it now...?

File: .github/workflows/pr_push.yml [hunk 5]
```diff
@@ -48,6 +48,31 @@ jobs:
...
+      - name: Install apt packages
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y cmake clang-format-15
+
+      - name: Configure CMake
+        run: >
+          cmake
+          -B ${{github.workspace}}/build
+          -DUMF_FORMAT_CODE_STYLE=ON
```
Comment (ldorau) [inline .github/workflows/pr_push.yml:69]: The coding style of the whole repository should be checked, so all components turned off by default should be also enabled:
- UMF_BUILD_LIBUMF_POOL_DISJOINT
- UMF_BUILD_LIBUMF_POOL_JEMALLOC
- UMF_BUILD_LIBUMF_POOL_SCALABLE
- UMF_BUILD_BENCHMARKS

Comment (ldorau) [inline .github/workflows/pr_push.yml:69]: The coding style check is so fast that - in my opinion - (apart from this build at the beginning that no doubt is needed) the coding style check could be added to all basic builds, just to be sure that coding style is verified in all possible configurations.

**Review by ldorau — CHANGES_REQUESTED** [review]

File: .github/workflows/pr_push.yml (hunk 5 above)
Comment (bratpiorka) [reply .github/workflows/pr_push.yml:69]: The coding style is checked for all h, hpp, c, cpp etc files in the project. There is no need to run it multiple times. If we decided to use separate workflows we should keep them short and  ideally each should be responsible for something different.

Comment (lukaszstolarczuk) [reply .github/workflows/pr_push.yml:69]: does it make any difference...? we use glob pattern to find files in repo for scanning

File: .github/workflows/pr_push.yml (hunk 4 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml:74]: Done

File: .github/workflows/pr_push.yml (hunk 1 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml:52]: Done

File: .github/workflows/pr_push.yml (hunk 4 above)
Comment (KFilipek) [reply .github/workflows/pr_push.yml:74]: Done

**Review by PatKamin — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

File: .github/workflows/pr_push.yml (hunk 5 above)
Comment (ldorau) [reply .github/workflows/pr_push.yml:69]: Right

Comment (ldorau) [reply .github/workflows/pr_push.yml:69]: Right

**Review by ldorau — APPROVED** [review]

**Pull request merged**